server/src/main/java/org/apache/druid/curator/discovery/CuratorDruidLeaderSelector.java

```diff
@@ -70,6 +71,8 @@ public CuratorDruidLeaderSelector(CuratorFramework curator, @Self DruidNode self
     // election until LeaderLatch.start() is called in registerListener(). This allows clients to observe the current
     // leader without being involved in the election.
     this.leaderLatch.set(createNewLeaderLatch());
+
+    curator.getConnectionStateListenable().addListener(this::handleConnectionStateChanged);
```

Comment:
I don't think we need this now since we are already recreating the latch when `notLeader` is called.
See https://github.com/apache/druid/pull/16528#discussion_r1623237401 for reference.